src/homework2/task7.py

```diff
@@ -0,0 +1,173 @@
+    # str_ = 'Ita dotf ni dyca nsaw ecc.'  # simple sample
+    s = input('Type any string you want to transform: ')
+    n = int(input('How many times would you like to do that? '))
+    for count_ in range(n):  # transform the string
+        s = s[::2] + s[1::2]
+    return ''.join(s)
+
+
+def valid_parentheses():
+    """Exercise 3: Valid Parentheses.
+
+    https://www.codewars.com/kata/52774a314c2333f0a7000688
+    Write a function called that takes a string of parentheses, and determines
+    if the order of the parentheses is valid. The function should return true
+    if the string is valid, and false if it's invalid.
+
+    Examples
+    "()"              =>  true
+    ")(()))"          =>  false
+    "("               =>  false
+    "(())((()())())"  =>  true
+
+    Constraints
+    0 <= input.length <= 100
+
+    Along with opening (() and closing ()) parenthesis, input may contain any
+    valid ASCII characters. Furthermore, the input string may be empty and/or
+    not contain any parentheses at all. Do not treat other forms of brackets
+    as parentheses (e.g. [], {}, <>).
+    """
+    print("Exercise 3: Valid Parentheses.")
+    str_ = input("Enter a String with Parentheses '()': ")
+    parenth = '()'  # for entry check
+    stack = []  # stack where parentheses will be saved
+    string_par = ''  # new string with just parentheses
+    for char in str_:  # get rid of junk and leave just parentheses
+        if char in parenth:
+            string_par += char
+    # general case
+    for par in string_par:
+        if par == '(':  # add any '(' in stack
+            stack.append(par)
+        elif stack:  # otherwise remove last '(' from stack
+            stack.pop(-1)  # if stack is not empty
+        else:  # case when string contains more ')' then '('
+            return False
+    if stack:  # case when string contains more '(' then ')'
+        return False
+    else:
+        return True  # case with a right parentheses or string w/o parentheses
+
+
+def hex_string_to_RGB():
+    """Exercise 4: Convert A Hex String To RGB.
+
+    https://www.codewars.com/kata/5282b48bb70058e4c4000fa7
+    When working with color values it can sometimes be useful to extract the
+    individual red, green, and blue (RGB) component values for a color.
+    Implement a function that meets these requirements:
+
+    Accepts a case-insensitive hexadecimal color string as its parameter
+    (ex. "#FF9933" or "#ff9933")
+    Returns an object with the structure {r: 255, g: 153, b: 51} where r, g,
+    and b range from 0 through 255
+
+    Note: your implementation does not need to support the shorthand form of
+    hexadecimal notation (ie "#FFF")
+
+    Example:
+    "#FF9933" --> {r: 255, g: 153, b: 51}
+    """
+    import re
+
+    print("Exercise 4: Convert A Hex String To RGB.")
+    str_ = input("Enter a hexadecimal color string with a following format"
+                 " (e.g. #FF0056): ")
+    if re.match(r'#[0-9a-fA-F]{6}', str_):
```

Comment:
why do we need this validation? according to the task description, we receive valid string.

Reply:
Basically the validation was done to mimic real case when input can be made with mistakes. But you're right, for synthetic task the validation is redundant. So it was removed from code.